File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] res = new int[2];
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        for(int i=0 ; i<nums.length ; i++){
            map.put(nums[i],i);

        }

        for(int i=0 ; i<nums.length ; i++){
            if (map.containsKey(target-nums[i]) && map.get(target-nums[i])!=i){
                return new int[]{i, map.get(target-nums[i])};
            }

        }

        return new int[]{-1, -1};
        
    }
}

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-1.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        res = 0
        for i in range(len(nums)):
            s = 0
            j = i
            while j < len(nums):
                s += nums[j]
                j += 1

                if s == k:
                    res += 1

        return res
